components/automate-cli/cmd/chef-automate/backup.go

```diff
@@ -1387,3 +1434,226 @@ func getBackupStatusAsCompletedOrFailed(backup *api.BackupTask) bool {
+
+		if automateBackupPath != osPath {
+			err := fmt.Errorf("discrepancy in the backup paths. All backup paths should point to the same location. Backup path configured in automate node: %s, path_repo configured in opensearch node: %s. Please check the paths listed and ensure to configure all backup paths to point to the same destination. ", automateBackupPath, osPath)
+			return false, err
+		}
+	}
+	if snapshotServiceBackupPath == "" {
+		return true, nil
+	}
+	if automateBackupPath != snapshotServiceBackupPath {
+		err := fmt.Errorf("discrepancy in the backup paths. All backup paths should point to the same location. Backup path configured in automate node: %s, indices of snapshot services: %s. Please check the paths listed and ensure to configure all backup paths to point to the same destination. ", automateBackupPath, snapshotServiceBackupPath)
+		return false, err
+	}
+	return true, nil
+}
+
+func compareSnapshotLocationOfServices(backupLocation string, sshUtil SSHUtil) (string, bool, error) {
+	var isEs5Service, isEs6Service, allowRestore bool
+	var snapshotEs5ServicePath, snapshotEs6ServicePath, complianceServicePath, eventFeedServicePath, erchefServicePath, ingestServicePath string
+	osSnapshot, err := getSnapshotLocationsOfServices(sshUtil)
+	if err != nil {
+		return "", false, err
+	}
+	if osSnapshot.Es6Compliance == nil && osSnapshot.Es6EventFeed == nil && osSnapshot.Es6Erchef == nil && osSnapshot.Es6Ingest == nil {
+		isEs6Service = false
+	} else {
+		isEs6Service = true
+	}
+	if osSnapshot.Es5Compliance == nil && osSnapshot.Es5EventFeed == nil && osSnapshot.Es5Erchef == nil && osSnapshot.Es5Ingest == nil {
+		isEs5Service = false
+	} else {
+		isEs5Service = true
+	}
+	if !isEs5Service && !isEs6Service {
+		logrus.Debug("no snapshots found in the cluster")
+		allowRestore = true
+		return "", allowRestore, nil
+	} else {
+		switch backupLocation {
+		case "fs":
+			if isEs6Service {
+				if osSnapshot.Es6Compliance.Type == backupLocation && osSnapshot.Es6Erchef.Type == backupLocation && osSnapshot.Es6EventFeed.Type == backupLocation && osSnapshot.Es6Ingest.Type == backupLocation {
+					complianceServicePath = getSnapshotPathDetails(osSnapshot.Es6Compliance.Settings.Location)
+					erchefServicePath = getSnapshotPathDetails(osSnapshot.Es6Erchef.Settings.Location)
+					eventFeedServicePath = getSnapshotPathDetails(osSnapshot.Es6EventFeed.Settings.Location)
+					ingestServicePath = getSnapshotPathDetails(osSnapshot.Es6Ingest.Settings.Location)
+					if complianceServicePath == erchefServicePath && erchefServicePath == eventFeedServicePath && erchefServicePath == ingestServicePath {
+						allowRestore = true
+						snapshotEs6ServicePath = complianceServicePath
+					}
+				} else {
+					err := fmt.Errorf("discrepancy in the backup types. Backup path of all indices should be %s. Refer `https://docs.chef.io/automate/ha_backup_restore_file_system/#troubleshooting` to do a clean up of the snapshot repo", backupLocation)
```

Comment:
This piece of code is repeated and probably can be made common in the conditions